### PR TITLE
Improve connection towards Azure

### PR DESCRIPTION
**Pull request by ace-e4s**
## Description
Establish and reuse `requests.Session` object in connection towards Azure. In this was, the underlying TCP connection is reused and potentially result in more performant and stable network connection. (See, https://docs.python-requests.org/en/latest/user/advanced/#session-objects)

The effect is marginal and barely observable in practice (normal office conditions), since there are other bottlenecks in the process. Nonetheless, an improvement of the code quality wrt network connection performance.

## Checklist
- [ ] PR title is descriptive and fit for injection into release notes (see tips below)
- [ ] Correct label(s) are used

PR title tips:
* Use imperative mood
* Describe the motivation for change, issue that has been solved or what has been improved - not how
* Examples:
  * Add functionality for Allan variance to sensor_4s.simulate
  * Upgrade to support Python 9.10
  * Remove MacOS from CI

### CODE REVIEW

File: datareservoirio/storage/storage.py [hunk 1]
```diff
@@ -15,6 +15,14 @@
 
 log = logging.getLogger(__name__)
 
+_AZURE_SESSION = requests.Session()
```
Comment (bjorn-einar-bjartnes-4ss) [inline datareservoirio/storage/storage.py:18]: Is this really a blobstorage or blobstorageaccount session? All of our resources are in azure.

Comment (branislav-jenco-4ss) [reply datareservoirio/storage/storage.py:18]: Do you mean we should call it `BLOBSTORAGE_SESSION` to be more precise? :)

**Comment by branislav-jenco-4ss** [comment]
Regarding retries over DNS failures, from what I read it should be handled, though I haven't tested.

**Comment by ace-e4s** [comment]
All low level network stuff  is handled by [urllib3](https://urllib3.readthedocs.io/en/stable/). Almost certain that fundamentals are covered.

File: datareservoirio/storage/storage.py (hunk 1 above)
Comment (ace-e4s) [reply datareservoirio/storage/storage.py:18]: Just keep the underscore prefix, and call it whatever you like :)

**Comment by ace-e4s** [comment]
Theoretically, we run at the risk of leaving a session (and all the related low level resources) open and allocated. But, since its one session per Python kernel, I'm sure it wont have an impact overall.

File: datareservoirio/storage/storage.py (hunk 1 above)
Comment (branislav-jenco-4ss) [reply datareservoirio/storage/storage.py]: renamed to BLOBSTORAGE_SESSION now

**Review by branislav-jenco-4ss — APPROVED** [review]

**Review by branislav-jenco-4ss — APPROVED** [review]

**Comment by branislav-jenco-4ss** [comment]
LGTM

**Review by bjartwolf — APPROVED** [review]

**Pull request merged**